tests/test_language_java.py

```diff
@@ -120,6 +120,58 @@ def test_remove_method():
         )
 
 
+def test_copy_and_rename_method():
+    with tempfile.TemporaryDirectory() as tempdir:
+        shutil.copyfile(
+            "tests/testdata/SampleClass.java", tempdir + "/SampleClass.java"
+        )
+
+        java.copy_and_rename_method(
+            tempdir + "/SampleClass.java", "public static void foo()", "foo", "foobar"
+        )
+        java.copy_and_rename_method(
+            tempdir + "/SampleClass.java", "public void asdf()", "asdf", "xyz"
+        )
+        assert_matches_golden(
+            "tests/testdata/SampleCopyMethodGolden.java", tempdir + "/SampleClass.java"
+        )
+
+
+def test_deprecate_method():
+    with tempfile.TemporaryDirectory() as tempdir:
+        shutil.copyfile(
+            "tests/testdata/SampleClass.java", tempdir + "/SampleClass.java"
+        )
+        DEPRECATION_WARNING = """  /*
+   * @deprecated This method will be removed in the next major version.
+   * Use {{@link #{new_method}()}} instead
+   */
+@Deprecated
+"""
```

Comment:
Usability wise, I would prefer if this parameter was just the deprecated javadoc message (`This method will be removed in the next major version....`. The `@Deprecated` annotation should be done by the method for me.

Reply:
Done